numba_dpex/examples/kernel/pipelining.py

```diff
@@ -0,0 +1,137 @@
+# SPDX-FileCopyrightText: 2020 - 2023 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
```

Comment:
Can you please add a module level docstring and any other docstring as needed explaining what the example shows. We have in the past users open issues asking for the examples to be documented. (see #319).